### ISSUE TITLE
Training in iOS

### ISSUE THREAD

**Issue by johanlantz** [pytorch/pytorch#29652]
Hi,

I am using your helloworld example in iOS and I am trying to add on device training and I got stuck pretty much right away when trying to create the optimizer.

The first issue was that the optim.h file was not found but I managed to correct that by adding ```"$(PODS_ROOT)/LibTorch/install/include/torch/csrc/api/include"``` to the header search path.

Now I am trying to create the optimized object similar to how it is done in the normal c++ example, something like:
```  torch::optim::SGD optimizer(_impl.get_parameters(), torch::optim::SGDOptions(0.01).momentum(0.5)); ````

Simply replacing model.parameters() with _impl.get_parameters() but this will not build in Xcode.

It might be a stupid question but I could not find any resources of running training on torch mobile so I am just trying to fit the pieces one by one.

Any suggestions on what I might be missing with this first step.

**Comment by dreiss** [comment]
Thanks for the report.  We're developing support for on-device training right now.  If you post the error message you get from XCode, we can try to point you in the right direction.  I just want to warn you that this will be a bumpy road.

**Comment by johanlantz** [comment]
@dreiss  Thanks for the quick reply and no worries about the ride being bumpy, that just makes it more fun.

My goal, for now, is quite basic. I just want to be able to train a model and pull out the updated weights and biases to run some tests with federated learning. I am completely new to torch (first day today) but it seems the parameters will give me that out of the box.

I ran this example:
https://github.com/goldsborough/examples/blob/cpp/cpp/mnist/mnist.cpp
on mac-os just fine and then I tried to just break out the training part and insert it into your HelloWorld example for iOS here: https://github.com/pytorch/ios-demo-app/tree/master/HelloWorld

so the first bump was that there was no model.parameters() so I assumed/guessed this would then be _impl.get_parameters() but when trying that with:
``` torch::optim::SGD optimizer(_impl.get_parameters(), torch::optim::SGDOptions(0.01).momentum(0.5));```
 Xcode complains that "No matching constructor for initialization of torch::optim::optimized

I am surely using it wrong.

Either way, if this explanation does not make any sense, just having the TorchBridge objective C class complemented with a training example would be tremendously helpful to get started.

Thanks in advance for your time
